src/snowflake/snowpark/table.py

```diff
@@ -369,6 +374,21 @@ def sample(
 
         from snowflake.snowpark.mock._connection import MockServerConnection
 
+        stmt = None
+        if _emit_ast:
+            # AST.
+            stmt = self._session._ast_batch.assign()
+            ast = with_src_position(stmt.expr.sp_table_sample, stmt)
+            if frac:
+                ast.probability_fraction.value = frac
```

Comment:
This looks surprising to me. In general, I think we used `build_expr_from_python_val` for this type of translation. We need to apply the same pattern to all values.

Reply:
This is the same code as the dataframe.sample, I can update that location as well.

Looking at the helper code, it seems similar but also sets some src_position information with the type.

Reply:
I tried this but it fails:

        elif isinstance(obj, float):
>           ast = with_src_position(expr_builder.float64_val)
E           AttributeError: float64_val

It looks like this is only used if it's a variant type.